### DISCUSSION TITLE
OTA Rauc install with Header in every request

### DISCUSSION THREAD

**Discussion by mwmike** [rauc/rauc#1279] [Q&A]
I have set up an image on an AWS S3 server that contains the RAUC bundle and Casync store. To access the S3 server, we use Cloudfront with a bearer token in the RAUC install header request. However, the issue we are facing is that the header is only presented in the first request and not in the subsequent ones. I was wondering if there is a way to present the header in each and every request?  If not is there another way that is recommended?

**Comment by jluebbe** [comment]
The chunk requests are done by casync itself, so that's not directly under RAUC's control. Which casync implementation do you use? Perhaps RAUC could be extended to pass (a subset of) the headers to casync?

**Reply by mwmike** [reply]
We are implementing RAUC on a Kirkstone image generated by Yocto using the meta-rauc repo. Rauc version 1.10.1.

**Reply by ejoerns** [reply]
The question was about the casync implementation. But since you mention meta-rauc I assume you use the 'normal' systemd-related casync from https://github.com/systemd/casync. 

Note that in https://rauc.readthedocs.io/en/latest/advanced.html#rauc-casync-support we also have a list of probably more actively supported casync variants.

**Comment by ejoerns** [comment]
@mwmike Did you try to use the RAUC's built-in [HTTP bundle streaming](https://rauc.readthedocs.io/en/latest/advanced.html#http-streaming) with AWS? Or have you encountered any problems with it?

**Reply by mwmike** [reply]
We are not using nbd in our build.  We will eventually shift to adaptive updates.

**Comment by piotrbulinski** [comment]
@mwmike did you consider passing the signature in the URL instead of relying on headers for auth? If the URL itself is signed, then you shouldn't have any issues when it's passed down to casync.

CloudFront [supports](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-signed-urls.html) signed URLs natively, so that should work.
